File: src/android/Messenger/app/src/main/java/com/example/prashant/messenger/TaskThreadItem.java
package com.example.prashant.messenger;

import java.io.IOException;
import java.net.Socket;

/**
 * Created by Prashant on 12/06/2017.
 */

public abstract class TaskThreadItem {
    Socket s;

    protected TaskThreadItem() throws IOException {
    }

    public abstract void doWork();

    public void setSocket(Socket socket) {
        s = socket;
    }
}

File: src/C#/Messenger/Messenger/TaskThreadItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;

namespace Messenger
{
    public abstract class TaskThreadItem
    {
        Socket s;

        public abstract void doWork();
        public void setSocket(Socket socket)
        {
            s = socket;
        }
    }
}
